sdk/core/core-util/src/isNode.ts

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * A constant that indicates whether the environment the code is running is Node.JS.
+ */
+export const isNode =
+  typeof process !== "undefined" && Boolean(process.version) && Boolean(process.versions?.node);
```

Comment:
I think it is not worth it but should we have a `isNode.browser.ts` that sets it to false?

Reply:
Yeah I wondered about this, like why is the check dynamic at all when we could solve it with bundling, but I seem to remember there was some scenario where we wanted to pull in the same code to run in both places (e.g. we couldn't know which bundle we were part of.)

Happy to patch this later

Reply:
It is often convenient to do a quick runtime check when you need to skip/do something simple in one platform or another. It could be solved by abstracting out into a function something like "doBehavior()" and override for node/browsers, but it's not super handy. In general I also feel like it's better to have one source asset that doesn't rely on bundling and browser overrides if you've just got one minor piece of customization you need, but I admit that we are not at all rigorous about that.


However, adding the browser.ts doesn't regress this story any since if you're bundling you get a nice quick check (for browser only).

Reply:
I'll add the browser.ts version but leave this code otherwise untouched